src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -242,54 +242,47 @@ public OperationalResponse favoriteMovie(OperationalRequest request) {
 		return new OperationalResponse(true);
 	}
 	
-	public OperationalResponse checkout(OperationalRequest request) {
+	public ResponseEntity<?> checkout(OperationalRequest request) {
 		int movieId = request.getMovieId();
 		int userId = request.getUserId();
+		Inventory record = null;
+		User user;
+		PrimaryKey pk = new PrimaryKey(userId, movieId);
 		OperationalResponse resp = new OperationalResponse(false);
-		Optional<Inventory> inventory = inventoryRepository.findById(movieId);
-		if(!inventory.isPresent()) {
-			//return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Movie with Id " + movieId + " does not exist!");
-			return resp;
+		if(!inventoryRepository.existsById(movieId)) {
+			HashSet<Integer> s = new HashSet<Integer>();
+			s.add(movieId);
+			SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(s);
+			if(searchAPIResp == null) {
+				resp.setMessage("Search API has no record for Movie with Id " + movieId);
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body(resp);
+			}
+			record = new Inventory(movieId, DEFAULT_NUM_OF_MOVIES, DEFAULT_NUM_OF_MOVIES);
 		}
-		Inventory record = inventory.get();
+		record = (record == null) ? inventoryRepository.getOne(movieId) : record;
```

Comment:
Can we do something like below? This way we will save 1 extra DB call. Also, `getOne(movieId)` throws unchecked exception `EntityNotFoundException` if the entity is not found. We should try and avoid any underlying code which throws unchecked exceptions. 

```
record = inventoryRepository.findById(movieId);
if(recode == null) {
    HashSet<Integer> s = new HashSet<Integer>();
    s.add(movieId);
    SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(s);
    if(searchAPIResp == null) {
        resp.setMessage("Search API has no record for Movie with Id " + movieId);
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(resp);
    }
    record = new Inventory(movieId, DEFAULT_NUM_OF_MOVIES, DEFAULT_NUM_OF_MOVIES);
}
```

Reply:
This is true. My bad. Thanks for pointing it out.